### PR TITLE
Make windows compilation successful

### LINKED ISSUE #1: Does not compile on Windows
On windows, using make that comes with mingw-w64 (i think), the compilation fails with error **'unistd.h' file not found**

Windows 10
make version 4.3
Clang version 11.0.0

EDIT :: Apparently my clang toolchain was incorrect, but now I'm missing poll.h instead of unistd.h

### PR DESCRIPTION
**Pull request by Kylogias**
NOTE :: This lays the groundwork for windows compilation to be successful but it doesn't ensure that the compiled code will run properly. This PR also cleans up some code to make MinGW-W64 throw less errors and to make time measurements more exact (1024*1024 for megahertz instead of 1000000). Should close #1

### CODE REVIEW

**Comment by Kylogias** [comment]
BUG :: Compilation _**IS**_ successful but when you try to run in the emulator, it silently fails

**Review by TheTechRobo — CHANGES_REQUESTED** [review]
I suggest putting the Windows build instructions inside the REAdME,along with eeything else.

**Comment by Kylogias** [comment]
> I suggest putting the Windows build instructions inside the REAdME,along with eeything else.

Done, although I don't know exactly what you mean when you say "everything else"

**Comment by jdah** [comment]
Wow, nice PR (and thank you for sticking in the style of the existing codebase!) don't have time to merge this in now, but will get to it in the next day or so :)

**Comment by Kylogias** [comment]
> Wow, nice PR (and thank you for sticking in the style of the existing codebase!) don't have time to merge this in now, but will get to it in the next day or so :)

Thanks, I literally didn't touch the codebase, I only touched the includes. I included stdio.h which iirc wasn't in the original code so you may want to check if it compiles

**Comment by CoolCat467** [comment]
Nice!

**Review by hdfsyu — APPROVED** [review]
YES YES YES

File: emu/emu.c [hunk 1]
```diff
@@ -27,11 +32,11 @@ static int run(struct JDH8 *state, const char *speed, char *out, usize n) {
 
     if (speed != NULL) {
         if (!strcasecmp(speed, "realtime")) {
-            hz = 100000000; // 100 MHz (fastest possible)
+            hz = 1024*1024*100; // 100 MHz (fastest possible)
```
Comment (jdah) [inline emu/emu.c:35]: This is not correct, Hertz are not powers of two, but ten.

File: Makefile-WIN [hunk 2]
```diff
@@ -0,0 +1,56 @@
+CC=gcc
+LD=gcc
+HXDP=xxd
+
+CCFLAGS=-std=c11 -O0 -g -Wall -Wextra -Wpedantic -Wstrict-aliasing -IC:/cdevlibs/include/ -v
```
Comment (jdah) [inline Makefile-WIN:5]: Are `cdevlibs` a universal thing for MinGW? doesn't look like it as far as I can tell.

Comment (jdah) [reply Makefile-WIN:5]: Ah, never mind, see in the README update.

File: emu/emu.c (hunk 1 above)
Comment (Kylogias) [reply emu/emu.c]: Fixed, although I kept the multiplication to keep it readable

**Comment by Kylogias** [comment]
I have identified a bug, although I am unsure how to fix it. The bug is that enter doesn't stop the simulation, at least on Windows

**Comment by jdah** [comment]
This has been merged in with other changes alongside cd6cf42a3e0e333939255c5fdbc5202d122e727e.

**Comment by jdah** [comment]
If you can fix the enter bug (that's all `poll` is used for btw) please open another PR :)

**Pull request closed without merging**